src/lib/support/BytesToHex.cpp

```diff
@@ -123,6 +89,40 @@ CHIP_ERROR BytesToHex(uint64_t src, char * dest_hex, size_t dest_size_max, BitFl
     return BytesToHex(buf, 8, dest_hex, dest_size_max, flags);
 }
 
+CHIP_ERROR MakeU8FromAsciiHex(const char * src, uint8_t * val)
+{
+    if (src == nullptr || val == nullptr)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    uint8_t ret = 0;
+    for (size_t i = 0; i < 2; ++i)
+    {
+        ret          = static_cast<uint8_t>(ret << 4);
+        char c       = src[i];
+        uint8_t cval = static_cast<uint8_t>(c);
+        if (c >= '0' && c <= '9')
+        {
+            ret = static_cast<uint8_t>(ret + cval - static_cast<uint8_t>('0'));
+        }
+        // Only uppercase is supported according to spec.
```

Comment:
This comment seems odd - what spec? Maybe remnant from other code? I believe we could just remove this.